addon/globalPlugins/webAccess/gui/properties.py

```diff
@@ -26,1011 +26,1031 @@
+
+		if res:
+			valProps = res[0].get_value()
+			val = ret(valProps) if valProps is not None else None
+			index = self.currentSelItem.GetId()
+			updateVal = self.getChoiceValueByKey(val, res[0].get_id())
+			self.listCtrl.SetItem(index, 1, updateVal)
+			return
+		return
+
+	def updateValueEvtToggle(self, evt):
+		"""
+		Event handler to update toggle button properties
+		"""
+		evtObj = evt.GetEventObject()
+		for p in self.propertiesList:
+			if evt.GetEventType() == wx.EVT_TOGGLEBUTTON.evtType[0]:
+				if evtObj.GetLabel() in p.get_displayName():
+					state = evtObj.GetValue()
+					if state:
+						p.set_value(True)
+						self.toggleBtn.SetValue(True)
+						self.updatePropertiesList(evtObj.GetLabel())
+					else:
+						p.set_value(False)
+						self.toggleBtn.SetValue(False)
+						self.updatePropertiesList(evtObj.GetLabel())
+					return
+
+	def updateValueEvtEdit(self, evt):
+		"""
+		Event handler updates Editable properties
+		"""
+		text = self.editable.GetValue()
+		index = self.currentSelItem.GetId()
+		item = self.getItemSelRow()
+		for p in self.propertiesList:
+			if self.customDisplayLabel(p) == item[0]:
+				p.set_value(text)
+		self.listCtrl.SetItem(index, 1, text)
+
+	def updateValueEvtChoice(self, evt):
+		"""
+		Event handler updates choice properties
+		"""
+		choiceVal = self.choice.GetString(self.choice.GetSelection())
+		selRow = self.getItemSelRow()
+		self.updateChoiceProperties(selRow[0], choiceVal)
+		forVal = filter(lambda x: x.get_displayName() == selRow[0], self.propertiesList)
+		res = list(forVal)
+		pos = self.choice.GetSelection()
+		retObjIncr = self.getIncDecObjById(res[0].get_id())
+		retObjIncr.setPos(pos + 1) if res[0].get_id() else None
+		self.updatePropertiesList(selRow[0])
+
+	@staticmethod
+	def buildAutoActionOptions(mgr):
+		actionsDict = mgr.getActions()
+		defaultval = ("Choose an option", "")
+		return [defaultval] + [(actionsDict[i], i) for i in actionsDict]
+
+	def getAutoActions(self):
+		"""
+		On init, function updates the autoActions list on run time
+		"""
+		mgr = self.context["webModule"].ruleManager
+		# Translator: State properties "unchecked"
+		self.autoActionOptions = self.buildAutoActionOptions(mgr)
+
+	@staticmethod
+	def buildMutationOptions():
+		defaultval = ("Choose an option", "")
+		return [defaultval] + [(mutationLabel, i) for i, mutationLabel in enumerate(mutationLabels)]
+
+
+	def getMutationOptions(self):
+		"""
+		On init function updates the run time mutation list
+		"""
+		self.mutationOptions = self.buildMutationOptions()
+
+
+	def getGestureOptions(self):
+		"""
+		On init function updates the run time gestures list
+		"""
+		self.gestures = []
+		defaultval = ("Choose an option", "")
+		self.gestures.insert(0, defaultval)
 
 
 class AppendListCtrl(ListControl):
-    """
-    Class append the context properties to the list
-    """
-    lst =[]
-    def __init__(self, clientListBox):
-        super(AppendListCtrl, self).__init__(clientListBox)
```

Comment:
We dropped support for Python 2.
No need any more to use this long form unless strictly necessary.
Please use Python 3 style `super().` instead in all new code.